src/goto-instrument/contracts/assigns.cpp

```diff
@@ -168,10 +168,35 @@ assigns_clauset::conditional_address_ranget::generate_snapshot_instructions()
   const auto skip_target =
     skip_program.add(goto_programt::make_skip(location_no_checks));
 
-  const auto validity_check_expr =
-    and_exprt{clean_guard,
-              all_dereferences_are_valid(guarded_slice.expr, parent.ns),
-              w_ok_exprt{guarded_slice.start_adress, guarded_slice.size}};
+  if(check_target_validity)
+  {
+    // We check that when guard condition is true,
+    // the target expression is either null or w_ok with the expected size.
+    // This assertion will be falsified whenever guarded_slice.start_adress
+    // is invalid.
+    auto target_validity_assertion =
+      instructions.add(goto_programt::make_assertion(
+        or_exprt{not_exprt{clean_guard},
+                 null_pointer(guarded_slice.start_adress),
+                 w_ok_exprt{guarded_slice.start_adress, guarded_slice.size}},
+        location_no_checks));
```

Comment:
May I suggest that you wrap this `or_exprt` in a call to `simplify_expr(..., parent.ns)`?

Reply:
fixed

Reply:
done